### PR TITLE
Add a command line interface

### PR DESCRIPTION
**Pull request by ghost**
This PR adds a simple command line interface to combine spec files. It takes any number of partial specs, and outputs a combined version to stdout.

### CODE REVIEW

**Comment by ghost** [comment]
Alright, I think this is done! I fixed it up to use safe_dump instead of dump, and added a test. I think the fixtures can probably be improved by splitting into different API instead of top level keys, but that's probably a job for another day.

File: src/specsynthase/specbuilder.py [hunk 1]
```diff
@@ -50,3 +50,6 @@ def add_spec(self, file_name):
         self._merge_part(self._parameters_key, parameters)
 
         return self
+
+    def dump(self, *args, **kwargs):
+        return yaml.safe_dump(*args, **kwargs)
```
Comment (allan-silva) [inline src/specsynthase/specbuilder.py]: There is no input data to safe_dump, I think that can be yaml.safe_dump(self, *args, **kwargs). As SpecBuilder has no Representer, maybe you  may want unpack dict into another dict like dict(**self).

File: tests/test_specsynthase.py [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+def test_dump():
+    base_dir = path.join(path.abspath(path.dirname(__file__)), "fixtures", "petstore")
+    parts = ["base.yml", "definitions.yml", "paths.yml"]
+    full = yaml.load(open(path.join(base_dir, "full.yml")).read())
+
+    spec = SpecBuilder()
+    for p in parts:
+        spec.add_spec(path.join(base_dir, p))
+
+    assert full == spec
```
Comment (allan-silva) [inline tests/test_specsynthase.py]: This test has no call to dump function.

File: tests/fixtures/petstore/paths.yml [hunk 3]
```diff
@@ -0,0 +1,352 @@
+paths:
```
Comment (allan-silva) [inline tests/fixtures/petstore/paths.yml:1]: Thanks for include this!

**Review by allan-silva — COMMENTED** [review]
Ben, thanks for contribution, is a great idea have a command line interface. In this way, spec-synthase no need always be a reference, users ca use it just to build a bigger spec.

I added some comments to PR, let me know if I misunderstood some thing.

**Comment by allan-silva** [comment]
BTW, I created an issue #3 and probably will create others, feel free to comment.

File: tests/test_specsynthase.py (hunk 2 above)
Comment (ghost) [reply tests/test_specsynthase.py]: Whoops! Obviously this will catch the problem with the code, too :).

**Comment by ghost** [comment]
Should be better now, sorry about that!

**Review by allan-silva — APPROVED** [review]

**Pull request merged**